modules/mobile/app/controllers/mobile/v0/payment_history_controller.rb

```diff
@@ -43,8 +39,26 @@ def bgs_service_response
       end
 
       def paginate(payments, validated_params)
+        available_years = payments.map { |p| p.date.year }.uniq.sort { |a, b| b <=> a }
+        start_date = validated_params[:start_date]
+        end_date = validated_params[:end_date]
+
+        unless start_date && end_date
+          most_recent_year = available_years.first
```

Comment:
Doesn't have any impact, but I think you could sort in ascending order above (i.e., without a block) and do .last here.